utils/mil_tools/mil_misc_tools/serial_tools.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+import serial
+
+
+class NoopSerial(serial.Serial):
+    '''
+    Inherits from serial.Serial, doing nothing for each function.
+    Allows super classes to implement custom behavior for simulating
+    serial devices.
+    '''
+    port = 'noop-serial'
+
+    def __init__(*args, **kwargs):
+        pass
+
+    def open(self):
+        pass
+
+    @property
+    def in_waiting(self):
+        return 0
+
+    @property
+    def out_waiting(self):
+        return 0
+
+    def close(self):
+        pass
+
+    def __del__(self):
+        pass
+
+    def read(self, **kwargs):
+        pass
+
+    def write(self, *args):
+        pass
+
+    def flush(self):
+        pass
+
+    def flushInput(self):
+        pass
+
+    def flushOuput(self):
+        pass
+
+    def reset_input_buffer(self):
+        pass
+
+    def reset_output_buffer(self):
+        pass
+
+    def send_break(self, *args, **kwargs):
+        pass
+
+
+class SimulatedSerial(NoopSerial):
```

Comment:
Great idea, allows for a lot of flexibility going forward. I don't recall us having a simulated serial device at all beforehand so this is good for testing out mission code before a pneumatic device is finished or any other serial device.